src/Command/ProcessCommand.php

```diff
@@ -211,6 +255,6 @@ private function getImages(
             $firstResult += $maxResults;
 
             $manager->clear();
-        } while ([] !== $images);
+        } while (($this->limitPerResource === null && [] !== $images) || $this->limitPerResource > $firstResult);
```

Comment:
```suggestion
        } while ([] !== $images && ($this->limitPerResource === null) || $this->limitPerResource > $firstResult);
```
I guess?

Reply:
No. If there's no limit, we check only by presence of images. If none are present we've reached the end, and stop looping. Otherwise, if we have a limit, we don't care about the presence of images.

Reply:
I am pretty sure you are right, I just don't get it. How I read your code is: If there are no more images to process, just keep on going if I said I wanted to process a certain amount

Reply:
Do while loops are always, always a mess to reason about :D

If the user doesnt specify a limit, the logic is the same as before (keep looping while the query returns images)

If a limit is specified, we don't care about if the query returned any images, we just care whether or not we've reached our limit (and `$firstResult` at this point is a counter, thanks to the logic in [here](https://github.com/Setono/SyliusImagePlugin/blob/7-process-command-control/src/Command/ProcessCommand.php#L239-L242) and [here](https://github.com/Setono/SyliusImagePlugin/blob/7-process-command-control/src/Command/ProcessCommand.php#L255))

Reply:
Actullay.. You are right 😁 🙇 We should not keep looping if we have a limit but no images. I'll take your correction :)